server/devdocs/bin/generate-proptypes-index.js

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env node
+
+/**
+ * This file generates an index of proptypes by component displayname, slug and folder name
+ */
+
+const fs = require( 'fs' ),
```

Comment:
Also, let's add an external dependencies docblock, and keep `root` and `pathSwap` separate.